src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -383,5 +383,25 @@ public static void ToDeserializeCall(this CodeWriter writer, ITypeProvider imple
             SerializationFormat.Duration_ISO8601 => "P",
             _ => null
         };
+
+        public static void WriteMethodDeserialization(this CodeWriter writer, JsonSerialization serialization, bool async,
```

Comment:
WriteResponseDeserialization.

Reply:
I called it 'method' deserialization because the 2 serialization contexts are: Method and Model. Meaning, you write serialization information into the client methods, and you also write serialization information for the models. So, if we would refactor, I think that split in naming makes the most sense.

Reply:
Yeah but it sounds like we're deserializing a method.

Reply:
It is like Baby Clothes. It is clothes for babies, not made out of babies. 👶 But, I'll change it to `WriteDeserializationForMethods` instead. Then, the others could be renamed to `WriteDeserializationForModels` in another PR.